tests/test_fill_sack_from_repos_in_cache.py

```diff
@@ -56,29 +59,34 @@ def _create_cache_for_repo(self, repopath, tmpdir):
         base.close()
 
     def _setUp_from_repo_path(self, original_repo_path):
-        self.tmpdir = tempfile.mkdtemp(prefix="dnf_test_")
-
-        self.repo_copy_path = os.path.join(self.tmpdir, "repo")
-        shutil.copytree(original_repo_path, self.repo_copy_path)
+        repo_copy_path = os.path.join(self.tmpdir, "repo")
+        shutil.copytree(original_repo_path, repo_copy_path)
 
-        self._create_cache_for_repo(self.repo_copy_path, self.tmpdir)
+        self._create_cache_for_repo(repo_copy_path, self.tmpdir)
 
         # Just to be sure remove repo (it shouldn't be used)
-        shutil.rmtree(self.repo_copy_path)
+        shutil.rmtree(repo_copy_path)
 
         # Prepare base for the actual test
         conf = dnf.conf.MainConf()
         conf.cachedir = os.path.join(self.tmpdir, "cache")
-        conf.installroot = os.path.join(self.tmpdir)
+        conf.installroot = self.tmpdir
+        conf.persistdir = os.path.join(conf.installroot, conf.persistdir.lstrip("/"))
+        conf.substitutions["arch"] = "x86_64"
+        conf.substitutions["basearch"] = dnf.rpm.basearch(conf.substitutions["arch"])
         self.test_base = dnf.Base(conf=conf)
         repoconf = dnf.repo.Repo(TEST_REPO_NAME, conf)
-        repoconf.baseurl = self.repo_copy_path
+        repoconf.baseurl = repo_copy_path
         repoconf.enable()
         self.test_base.repos.add(repoconf)
 
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix="dnf_test_")
+
```

Comment:
What about to add `self.test_base = None` to `setUp()`?

Reply:
It will prevent a crash when test will not call `self._setUp_from_repo_path()`

Reply:
Added, thank you.